server/planarserver.py

```diff
@@ -129,16 +122,140 @@ async def start_servers():
     print()
 
 
-main_app.on_shutdown.append(on_shutdown)
+def server_main(_args):
+    """Start the PlanarAlly server."""
 
+    if (not (FILE_DIR / "templates").exists()) and ("dev" not in sys.argv):
+        print(
+            "You must gather your par— you must build the client, before starting the server.\nSee https://www.planarally.io/server/setup/self-hosting/ on how to build the client or import a pre-built client."
+        )
+        sys.exit(1)
 
-if __name__ == "__main__":
+    mimetypes.init()
+    mimetypes.types_map[".js"] = "application/javascript; charset=utf-8"
+
+    save.check_save()
+
+    loop = asyncio.get_event_loop()
```

Comment:
I would revert this back to the top of the file.
The `if sys.platform == "win"` depends on the presence of this variable and moving that block into this function is not so pretty in my opinion.